packages/core/lib/console.js

```diff
@@ -264,27 +261,21 @@ class Console extends EventEmitter {
     const contextVars = {};
 
     abstractions.forEach(abstraction => {
+      const name = abstraction.contract_name;
       // don't overwrite Node's native objects - only load contracts
       // into the repl context when no conflict exists
-      if (
-        this.replContextNameConflicts[abstraction.contract_name].conflict ===
-        false
-      ) {
-        contextVars[abstraction.contract_name] = abstraction;
+      if (!this.knownReplNameConflicts.has(name)) {
+        contextVars[name] = abstraction;
       }
     });
 
-    if (this.warnUserOfConflicts) {
+    if (this.newReplNameConflicts.size > 0) {
       let contractNames = [];
-      for (const name in this.replContextNameConflicts) {
-        if (
-          this.replContextNameConflicts[name].conflict === true &&
-          this.replContextNameConflicts[name].warned === false
-        ) {
-          this.replContextNameConflicts[name].warned = true;
-          contractNames.push(name);
-        }
+      for (const name of this.newReplNameConflicts.entries()) {
```

Comment:
I think `this.newReplNameConflicts.keys()`, and `name` would be the actual name, so it wouldn't have to be dereferenced.

Reply:
Oh, we can go directly from the `Set` to a list by:
`[...this.newReplNameConflicts.keys()].join(', ')` which is the same as looping and collecting the name field in an array.

Reply:
Oh ok sure.

Reply:
`this.newReplNameConflicts` makes more sense to be a list since it's only used to collect and display (join) strings.

Reply:
I'm going to vote we leave it as a set as it has the api you originally requested (`clear`, `add`, etc.) and thus you will interact with it the same way as `replGlobals` and `knownNameConflicts`. I don't really think we gain much by having it be an array.